UI/src/application.js

```diff
@@ -35,6 +37,20 @@ var URLRouter = function(map, backend, models) {
         });
         map.setCenter(new OpenLayers.LonLat(response.middlePoint.x, response.middlePoint.y), 12);
       });
+    },
+
+    speedLimit: function(mmlId) {
+      applicationModel.selectLayer('speedLimit');
+      backend.getRoadLinkByMMLId(mmlId, function(response) {
+        eventbus.once('speedLimits:available', function() {
```

Comment:
`speedLimits:available` is triggered from `applicationModel.selectLayer` once speed limits are fetched. If speed limit fetch completes before `getRoadLinkByMMLId` this event handler will never be triggered and the flow fails.